base/io.jl

```diff
@@ -144,8 +144,8 @@ write(to::IO, p::Ptr) = write(to, convert(UInt, p))
 
 function write(s::IO, a::AbstractArray)
     nb = 0
-    for i in eachindex(a)
-        nb += write(s, a[i])
+    for b in a
+        nb += write(s, b)
```

Comment:
I was tempted to do that, but decided to try to blend in with the surrounding code. I'll reverse that decision and change it.